esphome/components/uart/uart_component.cpp

```diff
@@ -20,5 +20,18 @@ bool UARTComponent::check_read_timeout_(size_t len) {
   return true;
 }
 
+void UARTComponent::set_rx_full_threshold_ms(uint8_t time) {
+  uint8_t bytelength = this->data_bits_ + this->stop_bits_ + 1;
+  if (this->parity_ != UARTParityOptions::UART_CONFIG_PARITY_NONE)
+    bytelength += 1;
+  int32_t val = (this->baud_rate_ / (bytelength * 1000 / time)) - 1;
+  if (val < 1) {
+    val = 1;
+  } else if (val > 120) {
+    val = 120;
+  }
```

Comment:
`#include "esphome/core/helpers.h"` and use `clamp` here:
```suggestion
  int32_t val = clamp<int32_t>((this->baud_rate_ / (bytelength * 1000 / time)) - 1, 1, 120);
```